src/components/new-suggestion/areas/instruments/edit/instruments-list.tsx

```diff
@@ -0,0 +1,62 @@
+import { NewSuggestionInstrument } from "@/interfaces/new-suggestion"
+import InstrumentsListItem from "./instruments-list-item"
+import { AppDispatch, AppState } from "@/redux/store"
+import { useDispatch, useSelector } from "react-redux"
+import { useEffect, useState } from "react"
+import { updateNewSuggestion } from "@/redux/slices/new-suggestion.slice"
+
+const InstrumentsList = () => {
+  const dispatch: AppDispatch = useDispatch()
+  const newSuggestion = useSelector((state: AppState) => state.newSuggestion.suggestion)
+  const [instrumentListItems, setInstrumentListItems] = useState<NewSuggestionInstrument[]>(
+    newSuggestion.instruments
+  )
+
+  useEffect(() => {
+    setInstrumentListItems(newSuggestion.instruments)
+  }, [instrumentListItems, newSuggestion])
+
+  const onDeleteInstrument = (index: number): boolean => {
+    const newItems = [...instrumentListItems]
+    newItems.splice(index, 1)
+
+    dispatch(
+      updateNewSuggestion({
+        ...newSuggestion,
+        instruments: newItems,
+      })
+    )
+    return true
+  }
+
+  const onDescriptionChanged = (index: number, description: string) => {
+    const newItems = [...instrumentListItems]
+    newItems[index] = { ...newItems[index], description }
+
+    dispatch(
+      updateNewSuggestion({
+        ...newSuggestion,
+        instruments: newItems,
+      })
+    )
+
+    return true
```

Comment:
What are the returns for?

Reply:
I mostly write "on<ACTION>"functions with a boolean return for cases when it can fail later down the hierarchy. Can remove if preferred.

Reply:
Component A: component that calls onDescriptionChanged on a input element.
Component B: component that receives onDescriptionChanged from A, and passes it to C.
Component C: contains the logic that can allow the onDescriptionChanged to fail.

Component B can write logic for the case that onDescriptionChanged fails, even though the logic in C doesn't have that functionality yet.

onDescriptionChanged & other methods that I added we're early in my pull request, so the usecase didn't occur that i had to write that logic in Component B. As of now it has no function

Reply:
Aaah I seee. That makes sense. If it has no function now might as well remove it

Reply:
Resolved in new commit